pkg/controller/prune.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	"fmt"
+
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/golang/glog"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/brownfield"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/events"
+)
+
+type pruneFunc func(c *AppGwIngressController, appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, ingressList []*v1beta1.Ingress) []*v1beta1.Ingress
+
+var pruneFuncList = []pruneFunc{pruneNoPrivateIP, pruneProhibitedIngress}
+
+// PruneIngress filters ingress list based on filter functions and returns a filtered ingress list
+func (c *AppGwIngressController) PruneIngress(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext) []*v1beta1.Ingress {
+	prunedIngresses := cbCtx.IngressList
+	for _, prune := range pruneFuncList {
+		prunedIngresses = prune(c, appGw, cbCtx, prunedIngresses)
+	}
+
+	return prunedIngresses
+}
+
+func pruneProhibitedIngress(c *AppGwIngressController, appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, ingressList []*v1beta1.Ingress) []*v1beta1.Ingress {
+	if cbCtx.EnvVariables.EnableBrownfieldDeployment == "true" {
```

Comment:
We could avoid a stack allocation if we moved the `if cbCtx.EnvVariables.EnableBrownfieldDeployment == "true" {` up where we initialize `var pruneFuncList` -- the conditional can decide whether to `append` `pruneProhibitedIngress` to `pruneFuncList`